src/Simplify.elm

```diff
@@ -5998,6 +6004,69 @@ setFromListChecks =
         [ intoFnCheckOnlyCall (emptiableFromListChecks setCollection)
         , wrapperFromListSingletonChecks setCollection
         , onSpecificFnCallReturnsItsLastArgCheck Fn.Set.toList
+        , intoFnCheckOnlyCall
+            (\checkInfo ->
+                if checkInfo.expectNaN then
```

Comment:
Later improvement: I guess we could do this check but only for literals if `expectNaN` is `True`.